accepted/platform-intrinsics.md

```diff
@@ -14,30 +14,30 @@ consistent, imposes a performance penalty unacceptable to implementors seeking
 maximum app/service throughput.  To enable this last bit of performance
 improvement dotnet defines platform dependent intrinsics. This allows
 platform/hardware providers to define low level intrinsics that map to
-particular hardware features that are not consistent - i.e. target dependent.
+particular hardware features that are not available on all platforms.
 This document outlines the process for proposing and implementing these
 intrinsics. This process is intended to be open and can be initiated and
 implemented by any contributor or partner.
 
 ## Guidelines for Platform Dependent Intrinsics
 
-1. A platform intrinsic should expose a specific feature or semantic that is
-not in common with other platforms or hardware implementations.  
+1. A platform intrinsic should expose a specific feature or semantic that is not universally available on all platforms, and for which the implemented function is not easily recognizable by the JIT.  
    * If the functionality is common and performant - make it platform
-   independent. 
+   independent.
```

Comment:
I think I commented on this during the initial design as well, but it might be worth extending this section here.

There is some functionality that is common and performant but which, when exposed in a platform independent manner, loses some of the flexibility that makes it not suitable for use in all cases.

I think a good number of the vector intrinsics fall into this category. They are generally available on all architectures we support, are performant, and we exposed general functionality via the `System.Numerics.Vector` APIs. However, they lost some of their flexibility, required more complex (custom) support in the JIT, etc.

Additionally, even in the case where we know a general purpose API should be exposed for certain functionality and that functionality is generally supported on modern hardware (CRC32, BitOps, etc) there will need to be custom JIT support for the "fast-path". Exposing that custom support in the form of an intrinsic, and then having the general purpose API call the intrinsic, IMO, is the right way to do such APIs as it provides a general purpose API, but also gives direct access for users who may want or require it.

Reply:
@tannergooding 's suggestion is consistent with the push back I got for a common crc32 API.